backend/src/main/java/com/ethan5/service/FriendService.java

```diff
@@ -19,4 +22,13 @@ public void addFriend(FriendRequest req) {
                 .build();
         repository.save(friendship);
     }
+
+    public void deleteFriend(String id1, String id2) {
+        Optional<Friend> f = repository.findByUser1IdAndUser2Id(id1, id2);
```

Comment:
As mentioned in the below comment, friend relationships are assumed to be bidirectional. Thus, we should add the following:

`Optional<Friend> f1 = repository.findByUser1IdAndUser2Id(id1, id2);`
`Optional<Friend> f2 = repository.findByUser1IdAndUser2Id(id1, id2);`

`f1.isPresent(repository::delete);`
`f2.isPresent(repository::delete);`

Reply:
Adding :+1: